app/controllers/activities_controller.rb

```diff
@@ -1,6 +1,8 @@
 class ActivitiesController < ApplicationController
+  before_action :authenticate_user!
+
  def index
- 	sleep 0.7
+    sleep 0.7
     @activities = PublicActivity::Activity.order("created_at desc").where(recipient_id: current_user).where("owner_id not in (?)", current_user).all
   end
 end
```

Comment:
@sonalkr132 tried it still no luck.Added this line to ability.rb  `include PublicActivity::Model` But it is giving an error undefined method has_many.only using `include PublicActivity` gives same problem as before.doing a puts statement outputs as it is.